src/spatialdata/_core/_elements.py

```diff
@@ -0,0 +1,104 @@
+"""SpatialData elements."""
+from __future__ import annotations
+
+from collections import UserDict
+from collections.abc import Iterable
+from typing import Any
+from warnings import warn
+
+from dask.dataframe.core import DataFrame as DaskDataFrame
+from datatree import DataTree
+from geopandas import GeoDataFrame
+
+from spatialdata._types import Raster_T
+from spatialdata._utils import multiscale_spatial_image_from_data_tree
+from spatialdata.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels2DModel,
+    Labels3DModel,
+    PointsModel,
+    ShapesModel,
+    get_axes_names,
+    get_model,
+)
+
+
+class Elements(UserDict[str, Any]):
+    def __init__(self, shared_keys: set[str | None]) -> None:
+        self._shared_keys = shared_keys
+        super().__init__()
```

Comment:
```suggestion
class Elements(UserDict[str, Any]):
    _shared_keys = set
    def __init__(self, shared_keys: set[str | None]) -> None:
        super().__init__()
```
Would there be a downside to making `_shared_keys` a class attribute here?